### PR TITLE
NoReadyVirtController and NoReadyVirtOperator are never triggered 

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
The admin can edit deployment and cause the virt-controller/operator to not be ready.
As a consequence, NoReady* alerts would not fire. This PR fixes this case.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #https://bugzilla.redhat.com/show_bug.cgi?id=2036676

**Special notes for your reviewer**:
I have packaged multiple fixes into this PR. I will try to ping relevant people for reviews.

**Release note**:
```release-note
NoReadyVirtController and NoReadyVirtOperator should be properly fired.
```

### CODE REVIEW

File: hack/prom-rule-ci/verify-rules.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@
 source $(dirname "$0")/../common.sh
 
 fail_if_cri_bin_missing
-readonly PROM_IMAGE="docker.io/prom/prometheus:v2.15.2"
+readonly PROM_IMAGE="quay.io/prometheus/prometheus:v2.15.2"
```
Comment (dhiller) [inline hack/prom-rule-ci/verify-rules.sh:22]: Great find! :+1:

**Review by dhiller — COMMENTED** [review]
One question: I was a bit confused as I'd have expected an increased number here: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7529/pull-kubevirt-prom-rules-verify/1511702990237470720#1:build-log.txt%3A182

than from another random run here:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7103/pull-kubevirt-prom-rules-verify/1511632986519375872#1:build-log.txt%3A181

as you added tests. I guess that either the number is somehow misleading or your tests aren't executed for some reason. Can you check?

**Comment by dhiller** [comment]
> > One question: I was a bit confused as I'd have expected an increased number here: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7529/pull-kubevirt-prom-rules-verify/1511702990237470720#1:build-log.txt%3A182
> > than from another random run here: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7103/pull-kubevirt-prom-rules-verify/1511632986519375872#1:build-log.txt%3A181
> > as you added tests. I guess that either the number is somehow misleading or your tests aren't executed for some reason. Can you check?
> 
> Those are alerting rules, not tests.

Sure, but still I was expecting the number to change, as you added rules. So to clarify: the numbers that are printed are not the number of rules? If they aren't, what exactly to they count?

**Comment by dhiller** [comment]
Also, wouldn't it be possible to verify this with a unit test at least?

**Comment by xpivarc** [comment]
> > > One question: I was a bit confused as I'd have expected an increased number here: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7529/pull-kubevirt-prom-rules-verify/1511702990237470720#1:build-log.txt%3A182
> > > than from another random run here: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7103/pull-kubevirt-prom-rules-verify/1511632986519375872#1:build-log.txt%3A181
> > > as you added tests. I guess that either the number is somehow misleading or your tests aren't executed for some reason. Can you check?
> > 
> > 
> > Those are alerting rules, not tests.
> 
> Sure, but still I was expecting the number to change, as you added rules. So to clarify: the numbers that are printed are not the number of rules? If they aren't, what exactly to they count?

The number represents rules. This PR is only adjusting the `records` we use and is adding tests for the cases described in PR.

**Review by enp0s3 — COMMENTED** [review]
@xpivarc Hi. Thank you for the fix. Have you managed to validate that the alert indeed being fired? I was under the impression that the ready gauge metric goes down just before the process exits, and the pod actually restarted. 

I also think that the release note is redundant since its a bugfix, WDYT?

**Comment by xpivarc** [comment]
> @xpivarc Hi. Thank you for the fix. Have you managed to validate that the alert indeed being fired? I was under the impression that the ready gauge metric goes down just before the process exits, and the pod actually restarted.

I think it is only the best effort as the prometheus might not have enough time to scrape the metric(correct me if I am wrong). I will try to provision a cluster and have a look.
 
> I also think that the release note is redundant since its a bugfix, WDYT?

I think we use to include bugfixes in release notes.

**Comment by xpivarc** [comment]
@enp0s3 I can confirm this fixes the problem.

**Comment by xpivarc** [comment]
/hold 
Test failures are unrelated. An investigation is ongoing.

**Comment by dhiller** [comment]
Looks like the operator lanes stopped failing after the merge, retesting will show us.

**Pull request merged**